Installation scripts for daemon mode on Ubuntu using Upstart (resubmission)

msgilligan opened the pull request:
Contains configuration files and an installation script for setting up Master Core (`bitcoind`) in daemon mode on Ubuntu using the Upstart init daemon. This is part of the work necessary to implement Issue #131, but does not address the actual packaging.

Some of the files are from the current [upstream](https://github.com/bitcoin/bitcoin) `master` branch. The files in `contrib/msc-ubuntu` were created specifically for Master Core (a README and a default `bitcoin.conf` file for Master Core)

Here are some of the merged upstream PR's included in this "back-port" to Master Core.
- [bitcoin #4611](https://github.com/bitcoin/bitcoin/pull/4611)
- [bitcoin #4577](https://github.com/bitcoin/bitcoin/pull/4577)

Upstream issues:
- [bitcoin #4124](https://github.com/bitcoin/bitcoin/issues/4124)

Note: This is a resubmitted PR, the original PR was #129. This PR is now based off of the `mscore-0.0.8` branch and includes a [change](https://github.com/msgilligan/mastercore/commit/440f19274d63b5dfe2599ac5f88b40bc42209aa7) suggested by @dexX7.

ajweiss committed:
Add init scripts and docs for Upstart and OpenRC

(cherry picked from commit 234bfbf6a5fcba37e510e9cb6c1f2a629cd0290e)

Conflicts:
	contrib/init/bitcoind.service
	doc/README.md

msgilligan committed:
Ubuntu/Upstart init scripts for Master Core

Back-ported some files from BitCoin core (approved) PR’s with some
additions for Master Core.

See:
https://github.com/bitcoin/bitcoin/issues/4124
https://github.com/bitcoin/bitcoin/pull/4611

msgilligan committed:
Add Install instructions to README.md

msgilligan committed:
Also install bitcoin-cli into /usr/bin

msgilligan committed:
Documentation improvements

msgilligan committed:
Merge branch 'michael-0921' into msgilligan-msc-upstart

msgilligan committed:
Merge remote-tracking branch 'msc/mscore-0.0.8' into msgilligan-msc-upstart

msgilligan committed:
Comment out paytxfee setting

Use the default hardcoded in the source.

msgilligan committed:
Merge branch 'mscore-0.0.8' into msgilligan-msc-upstart

msgilligan committed:
Copied from ../init/bitcoind.conf

msgilligan committed:
Customize bitcoind.conf for Master Core

* remove the ‘-disablewallet’ flag
* Add parenthetic “(Master Core)” to description

* change install script to copy from ‘msc-ubuntu’ directory.

dexX7 commented:
@m21 @zathras-crypto: this was resubmitted and adjusted a few times to catch up, so I think this should be looked at rather sooner than later.

@msgilligan: is there anything you'd like me to test, so I can confirm correct behavior?

msgilligan commented:
@dexX7 You could test the install procedure that's documented in `contrib/msc-ubuntu/README.md`. You could then verify that bitcoind starts up on OS reboot and that you can kill bitcoind and it will restart.

One issue with the installation script as it stands now is that it is not intended to be an update script that will add init support to existing installations. @achamely brought that to my attention yesterday.

msgilligan commented:
These scripts will need to be changed now that the binaries are named `mastercored` and `mastercore-cli`. 

And we should also decide if we want to use the `bitcoin` user:group and all the same directory and file names as Bitcoin uses. I suppose that as long as our code defaults to using the `*bitcon*` names our installation scripts/package should probably do the same.

msgilligan committed:
Merge branch 'mscore-0.0.8' into msgilligan-msc-upstart

msgilligan committed:
Rename bitcoind to mastercored

… and bitcoin-cli to mastercore-cli

msgilligan committed:
Rename the upstart service to ‘mastercored’

But continue to user the bitcoin/bitcoind names for files, directories,
users & groups.

msgilligan committed:
Remove extra -cli from generated file name

msgilligan committed:
/etc/init file needs to be mastercored.conf

msgilligan committed:
Merge branch 'mscore-0.0.8' into msgilligan-msc-upstart

msgilligan committed:
Update instructions

* Reflect bitcoin -> mastercore name change
* Add instructions for starting the mastercored service

msgilligan committed:
Merge branch 'mscore-0.0.8' into msgilligan-msc-upstart

msgilligan committed:
rpcthreads=64 in default mastercore bitcoin.conf

Workaround for omniwallet issue #1075
https://github.com/mastercoin-MSC/omniwallet/issues/1075

msgilligan committed:
Merge branch 'mscore-0.0.8' into msgilligan-msc-upstart

whitj00 commented:
These should be changed to reflect omni
